### PR TITLE
Use Block objects instead of config arrays

### PR DESCRIPTION
**Pull request by lukecarbis**

### CODE REVIEW

File: php/helpers.php [hunk 1]
```diff
@@ -132,7 +131,7 @@ function block_value( $name ) {
  */
 function block_config() {
 	global $block_lab_config;
-	return $block_lab_config;
+	return (array) $block_lab_config;
```
Comment (lukecarbis) [inline php/helpers.php:134]: Returning as an array for backwards compatibility.

File: php/blocks/class-loader.php [hunk 2]
```diff
@@ -191,22 +186,6 @@ public function get_block_attributes( $block ) {
...
-				$attributes[ $field_name ]['meta'] = $field['meta'];
-			}
-
-			if ( ! empty( $field['selector'] ) ) {
-				$attributes[ $field_name ]['selector'] = $field['selector'];
-			}
-
-			if ( ! empty( $field['query'] ) ) {
-				$attributes[ $field_name ]['query'] = $field['query'];
-			}
```
Comment (lukecarbis) [inline php/blocks/class-loader.php:209]: These fields are literally never set, and don't actually do anything. Not sure how they ended up here.

Comment (kienstra) [reply php/blocks/class-loader.php:209]: Yeah, interesting.

**Comment by kienstra** [comment]
Hi @lukecarbis,
This looks good enough to proceed. I haven't reviewed it very closely, but the overall approach looks good.

**Comment by lukecarbis** [comment]
@kienstra Ready for your full review.

**Comment by kienstra** [comment]
Thanks, @lukecarbis!

File: php/blocks/class-block.php [hunk 3]
```diff
@@ -111,19 +126,14 @@ public function from_json( $json ) {
...
 		if ( isset( $config['fields'] ) ) {
-			foreach ( $config['fields'] as $field ) {
-				$field_defaults = array( 'name', 'label', 'control', 'type', 'order' );
-				$field_settings = array_diff( array_keys( $field ), $field_defaults );
-				foreach ( $field_settings as $setting ) {
-					$field['settings'][ $setting ] = $field[ $setting ];
-				}
-				$this->fields[] = new Field( $field );
+			foreach ( $config['fields'] as $key => $field ) {
+				$this->fields[ $key ] = new Field( $field );
```
Comment (kienstra) [inline php/blocks/class-block.php:130]: Nice change here! It's nice how it moves the Field-related logic into `Field::from_array()`

**Review by kienstra — APPROVED** [review]
**Approved**

Hi @lukecarbis,
This looks good.

The Field logic is moved to the `Field` class, and the Location column doesn't appear anymore:

location-column-removed-here

**Pull request merged**